lib/SchemaConnector.js

```diff
@@ -0,0 +1,249 @@
+   * @example
+   * // Use Firebase Cloud Firestore
+   * smartapp.contextStore(new FirestoreDBContextStore(firebaseServiceAccount, 'app-table-name'))
+   * @returns {SchemaConnector} SmartApp instance
+   */
+  callbackStore(value) {
+    this._contextStore = value;
+    return this
+  }
+
+  manufacturerName(value) {
+    this._manufacturerName = value;
+    return this
+  }
+
+  enableEventLogging(jsonSpace = null, enableEvents = true) {
+    this._enableEventLogging = enableEvents;
+    this._eventLoggingSpace = jsonSpace;
+    return this
+  }
+
+  /**
+   * Use with an AWS Lambda function. No signature verification is required.
+   *
+   * @param {*} event
+   * @param {*} context
+   */
+  async handleLambdaCallback(event, context) {
+    try {
+      const response = await this._handleCallback(event);
+      if (response.isError()) {
+        console.log("ERROR: %j", err);
+        return context.fail(response)
+      }
+      else {
+        return context.succeed(response);
+      }
+    }
+    catch (err) {
+      console.log("ERROR: %j", err);
+      return context.fail(err)
+    }
+  }
+
+  /**
+   * Use with a standard HTTP webhook endpoint app. Signature verification is required.
+   *
+   * @param {*} req
+   * @param {*} res
+   */
+  async handleHttpCallback(req, res) {
+    try {
+      const response = await this._handleCallback(req.body);
+      if (response.isError()) {
+        console.log("ERROR: %j", response);
+        res.status(500).send(response)
+      }
+      else {
+        res.send(response)
+      }
+    }
+    catch (err) {
+      console.log("ERROR: %j", err);
+      res.status(500).send(err)
+    }
+  }
+
+  async handleMockCallback(body) {
+    try {
+      return await this._handleCallback(body)
+    }
+    catch (err) {
+      return err;
+    }
+  }
+
+  async _handleCallback(body) {
+    if (this._enableEventLogging) {
+      console.log(`REQUEST ${JSON.stringify(body, null, this._eventLoggingSpace)}`)
+    }
+    if (body.headers) {
+      let response;
+      switch (body.headers.interactionType) {
+        case "discoveryRequest":
+          response = new DiscoveryResponse(body.headers.requestId);
+          await this._discoveryHandler(body.authentication.token, response, body);
+          break;
+
+        case "commandRequest":
+          response = new CommandResponse(body.headers.requestId);
+          await this._commandHandler(body.authentication.token, response, body.devices, body);
+          break;
+
+        case "stateRefreshRequest":
+          response = new StateRefreshResponse(body.headers.requestId);
+          await this._stateRefreshHandler(body.authentication.token, response, body);
+          break;
+
+        case "grantCallbackAccess":
+          response = new STBase(body.headers.requestId);
+          if (this._callbackAccessHandler) {
+            if (body.callbackAuthentication.clientId === this._clientId ) {
+
+              const grantResponse = await (
```

Comment:
Should probably break this into 2 statement for clarity:

const request = new AccessTokenRequest(
                  this._clientId,
                  this._clientSecret,
                  body.headers.requestId
                );
await request..getCallbackToken(
                  body.callbackUrls.oauthToken,
                  body.callbackAuthentication.code
                );

Reply:
:thumbsup: